src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/SunPKCS11.java

```diff
@@ -1507,6 +1520,15 @@ public Object newInstance0(Object param) throws
                     throw new NoSuchAlgorithmException("Unsupported algorithm: "
                             + algorithm);
                 }
+            } else if (type == KDF) {
+                try {
+                    return new P11KDF(token, algorithm, (KDFParameters) param,
```

Comment:
I'd expect `mechanism` before `param` as mechanism is needed for all services but `param` may not. Can we adjust the ordering here?

Reply:
Yes, we can swap them.